File: amic-platform/src/modules/ma/types/nda_markup.ts
// NDA 마크업 버전 타입

export interface NdaMarkup {
  id: string;
  nda_id: string;
  version_label: string;
  version_number: number;
  version_date: string;
  source_party: string | null;
  markup_type: string | null;
  file_name: string | null;
  file_size_bytes: number | null;
  changes_summary: string | null;
  key_changes: unknown[] | null;
  redline_issues_count: number | null;
  base_version_id: string | null;
  created_by_email: string | null;
  created_at: string;
  updated_at: string;
  has_file: boolean;
  has_redline: boolean;
}

export interface NdaMarkupListResponse {
  items: NdaMarkup[];
  total: number;
  limit: number;
  offset: number;
}

File: deal-mgmt/app/schemas/nda_markup.py
"""NDA 마크업 버전 스키마."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class NdaMarkupOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nda_id: uuid.UUID
    attachment_id: uuid.UUID | None = None
    version_label: str
    version_number: int
    version_date: str
    source_party: str | None = None
    markup_type: str | None = None
    file_name: str | None = None
    file_size_bytes: int | None = None
    changes_summary: str | None = None
    key_changes: list[Any] | None = None
    redline_issues_count: int | None = None
    base_version_id: uuid.UUID | None = None
    created_by_email: str | None = None
    created_at: datetime
    updated_at: datetime

    has_file: bool = False
    has_redline: bool = False

    @classmethod
    def model_validate(cls, obj: Any, **kwargs: Any) -> NdaMarkupOut:  # type: ignore[override]
        instance = super().model_validate(obj, **kwargs)
        if hasattr(obj, "file_path"):
            instance.has_file = bool(obj.file_path)
        if hasattr(obj, "redline_file_path"):
            instance.has_redline = bool(obj.redline_file_path)
        return instance


class NdaMarkupListResponse(BaseModel):
    items: list[NdaMarkupOut]
    total: int
    limit: int
    offset: int
